lib/generateMarkdown.js

```diff
@@ -243,7 +280,7 @@ function getPropertySummary(property, suppressWarnings) {
     var description = property.description;
 
     var required;
-    if (defined(property.required) && (property.required)) {
+    if (defined(property.required) && typeof property.required === 'boolean' && (property.required)) {
```

Comment:
I wonder if this is related to adding object recursion.  I will update the PR with the schema I was using once I've made sure it's sanitized.